src/anthropic/lib/bedrock/_client.py

```diff
@@ -64,6 +67,29 @@ def _prepare_options(input_options: FinalRequestOptions) -> FinalRequestOptions:
     return options
 
 
+def _infer_region() -> str:
+    """
+    Infer the AWS region from the environment variables or
+    from the boto3 session if available.
+    """
+    aws_region = os.environ.get("AWS_REGION")
+    if aws_region is None:
+        try:
+            import boto3
+
+            session = boto3.Session()
```

Comment:
we also create a session in _auth.py so ideally we'd reuse it as they can be expensive